lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/net.rb

```diff
@@ -461,6 +465,7 @@ def cmd_portfwd(*args)
 
             # Start the local TCP reverse relay in association with this stream
             relay = service.start_reverse_tcp_relay(channel,
+              'LocalHost'         => channel.params.localhost,
```

Comment:
This in my tests on a host that supports IPv4 and IPv6 is returning only the IPv6 address:

![image](https://user-images.githubusercontent.com/63261883/211388631-6727f562-0201-4bc3-9c97-307de04e7300.png)

This explains the new behavior of binding by default to the IPv6 address vs the IPv4 address.

Reply:
Further expansion of this shows that it seems like the `v6` flag is controlling this setting, and tracing this via the call to `Rex::Socket::Parameters.new` shows that we have the `lib/ruby/gems/3.0.0/gems/rex-socket-0.1.43/lib/rex/socket/parameters.rb` file on line 200 which is using the following logic to set this flag:

```
# Whether to force IPv6 addressing
    if hash['IPv6'].nil?
      # if IPv6 isn't specified and at least one host is an IPv6 address and the
      # other is either nil, a hostname or an IPv6 address, then use IPv6
      self.v6 = (Rex::Socket.is_ipv6?(self.localhost) || Rex::Socket.is_ipv6?(self.peerhost)) && \
        (self.localhost.nil? || !Rex::Socket.is_ipv4?(self.localhost)) && \
        (self.peerhost.nil? || !Rex::Socket.is_ipv4?(self.peerhost))
    else
      self.v6 = hash['IPv6']
    end
```

Reply:
For confirmation IPv4 doesn't work:

![image](https://user-images.githubusercontent.com/63261883/211399585-ae5c9707-b499-4185-904b-68ce92dfea9c.png)

![image](https://user-images.githubusercontent.com/63261883/211399675-4952aa41-97e2-4347-b043-a199380beba5.png)

![image](https://user-images.githubusercontent.com/63261883/211399804-fcd66e39-ce61-41e1-91b7-14aa10600719.png)

But IPv6 works:

![image](https://user-images.githubusercontent.com/63261883/211400906-a0e92e1c-8690-4618-b1ad-318ec6cb294d.png)

![image](https://user-images.githubusercontent.com/63261883/211400954-a7e9d38e-3c67-42ff-b778-0d66463c98dc.png)

Reply:
So I think what you're running into is what I was mentioning here:

> There is a small bug where if the host to listen on is an empty string (-r "" vs -r "0.0.0.0") it'll fail but if an address is specified, it will be used. I'll submit a patch to fix the empty string issue along with the Windows changes.

When I first opened the PR, the LocalHost value was used so everything was all good. Now that it's not though, this is highlighting the discrepancy in how Python opens TCP listening sockets. Python does *not* support dual stack so when the LocalHost TLV is omitted or it's an empty string, it does... weird things. You get what you're seeing on Windows, while it straight up fails on Linux.

I'll go ahead and submit a patch to the Python Meterpreter so it supports dual stack mode when available. That will make it listen on `0.0.0.0` and `::`, just note that it'll report itself as listening on `::`. The Windows Meterpreter will do this too once https://github.com/rapid7/metasploit-payloads/pull/599 is landed.